hs_composite_resource/models.py

```diff
@@ -257,14 +259,13 @@ def _create_xml_docs_for_folder(self, folder):
         # note: we can't get to all filesets from resource files since
         # it is possible to have filesets without any associated resource files
         res_file_objects = ResourceFile.list_folder(self, folder)
-        aggregations = []
-        for res_file in res_file_objects:
-            if res_file.has_logical_file and res_file.logical_file not in aggregations:
-                aggregation = res_file.logical_file
-                aggregations.append(aggregation)
-                if not aggregation.is_fileset:
-                    if aggregation.metadata.is_dirty:
-                        aggregation.create_aggregation_xml_documents()
+        logical_files = [res_file.logical_file for res_file in res_file_objects if
```

Comment:
set here

Reply:
perhaps abstract the building of this set into a function and reuse above too.  It'll help explain what it's doing in the code as well.

Reply:
Created a function.